eth-brownie/brownie
# Vyper nonpayable heuristic

### What I did
Adjust the heuristic for nonpayable reverts in vyper 0.2.11

Related issue: #

### How I did it
In Vyper 0.2.11 the nonpayable check no longer always happens immediately upon entering the contract, it instead happens after the last payable function. I've adjusted the logic for the heuristic to be a bit looser, it checks any `REVERT` without the source map instead of only at the very beginning of the contract.

### How to verify it
Confirm the tests are now passing. This is the reason for failing CI on several open PRs.
